Sources/TensorFlow/Operators/NN.swift

```diff
@@ -138,434 +137,421 @@ public extension Padding {
-            padding: padding.raw)
-    }
+@usableFromInline
+func _vjpAvgPool2D<Scalar: TensorFlowFloatingPoint>(
+    _ x: Tensor<Scalar>,
+    filterSize: (Int, Int, Int, Int),
+    strides: (Int, Int, Int, Int),
+    padding: Padding
+) -> (Tensor<Scalar>, (Tensor<Scalar>) -> Tensor<Scalar>) {
+    // TODO: Currently this is not higher order differentiable. Redefine in
+    // closed form.
+    let value = avgPool2D(x, filterSize: filterSize, strides: strides, padding: padding)
+    return (value, { v in
+        Raw.avgPoolGrad(
+            origInputShape: x.shapeTensor,
+            grad: v,
+            ksize: [Int32(filterSize.0), Int32(filterSize.1),
+                    Int32(filterSize.2), Int32(filterSize.3)],
+            strides: [Int32(strides.0), Int32(strides.1),
+                      Int32(strides.2), Int32(strides.3)],
+            padding: padding.raw
+        )
+    })
+}
 
-    /// Computes a 3-D average pooling, with the specified kernel sizes, strides,
-    /// and padding.
-    ///
-    /// - Parameters:
-    ///     - kernelSize: The dimensions of the pooling kernel.
-    ///     - strides: The strides of the sliding filter for each dimension of the
-    ///         input.
-    ///     - padding: The padding for the operation.
-    @inlinable @inline(__always)
-    @differentiable(
-        wrt: self, vjp: _vjpAveragePooled3D(kernelSize:strides:padding:)
-        where Scalar : TensorFlowFloatingPoint
-    )
-    func averagePooled3D(
-        kernelSize: (Int, Int, Int, Int, Int),
-        strides: (Int, Int, Int, Int, Int),
-        padding: Padding
-    ) -> Tensor {
-        return Raw.avgPool3D(
-	    self,
-            ksize: [Int32(kernelSize.0), Int32(kernelSize.1),
-                    Int32(kernelSize.2), Int32(kernelSize.3), Int32(kernelSize.4)],
+@usableFromInline
+func _vjpAvgPool3D<Scalar: TensorFlowFloatingPoint>(
+    _ x: Tensor<Scalar>,
+    filterSize: (Int, Int, Int, Int, Int),
+    strides: (Int, Int, Int, Int, Int),
+    padding: Padding
+) -> (Tensor<Scalar>, (Tensor<Scalar>) -> Tensor<Scalar>) {
+    // TODO: Currently this is not higher order differentiable. Redefine in
+    // closed form.
+    let value = avgPool3D(x, filterSize: filterSize, strides: strides, padding: padding)
+    return (value, { v in
+        return Raw.avgPool3DGrad(
+            origInputShape: x.shapeTensor,
+            grad: v,
+            ksize: [Int32(filterSize.0), Int32(filterSize.1), Int32(filterSize.2),
+                    Int32(filterSize.3), Int32(filterSize.4)],
             strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3),
                       Int32(strides.4)],
-            padding: padding.raw)
-    }
+            padding: padding.raw
+        )
+    })
 }
+
+
+/// Computes a 2-D convolution using `self` as input, with the specified
+/// filter, strides, and padding.
+///
+/// - Parameters:
```

Comment:
```suggestion
/// - Parameters:
///   - x: The input.
```
Same for other public APIs that were renamed.

Reply:
Will make the same change everywhere else where necessary.